tooling/artifact_cli/src/cli/execute_cmd.rs

```diff
@@ -0,0 +1,220 @@
+use std::{collections::BTreeMap, path::PathBuf};
+
+use acir::{circuit::Program, native_types::WitnessStack, FieldElement};
+use bn254_blackbox_solver::Bn254BlackBoxSolver;
+use clap::Args;
+use color_eyre::eyre::{self, bail};
+
+use nargo::{foreign_calls::DefaultForeignCallBuilder, NargoError, PrintOutput};
+use noir_artifact_cli::{
+    errors::CliError,
+    fs::{inputs::read_inputs_from_file, witness::save_witness_to_dir},
+    Artifact,
+};
+use noirc_abi::{input_parser::InputValue, Abi};
+use noirc_artifacts::debug::DebugArtifact;
+
+use super::parse_and_normalize_path;
+
+/// Execute a binary program or a circuit artifact.
+#[derive(Debug, Clone, Args)]
+pub(crate) struct ExecuteCommand {
+    /// Path to the JSON build artifact (either a program or a contract).
+    #[clap(long, short, value_parser = parse_and_normalize_path)]
+    artifact: PathBuf,
```

Comment:
I'd make this a positional argument as we're always going to need it.

Reply:
We currently have a mixed approach:
* `inspector` commands take it as a positional arg 
* `profiler` commands take `--artifact-path`
* `acvm` takes `--bytecode` 
* this one takes `--artifact` (I don't like `--path` because we use e.g. `--target-dir` not `--target-dir-path`).

I personally find it more readable when we use options. They can be made mandatory, but it's more self documenting. When I see `nargo execute witness` I'm always confused what witness we're trying to execute. 

If we saw this command invoked, it already takes quite a few paths: `prover_file`, `output_dir`, `oracle_file`, `oracle_root_dir`, and the `artifact`. If these are long paths, the explicit naming would be helpful IMO.

Reply:
FWIW I renamed it to `--artifact-path` to match the profiler, which appears in some scripts in `aztec-packages`. 

I tried `#[arg, value_parser = parse_and_normalize_path)]` and it seemed happy with it, so if you think positional is the way to go, I'll change it. But just to note that as `Usage: noir-artifact execute [OPTIONS] --prover-file <PROVER_FILE> <ARTIFACT>` shows, `--prover-file` is also mandatory for `execute`, and I have memories of commands like `whatever execute --something 1 --other-thing 2 --yet-another /long/path/to/some/sub/directory /another-long/path/to/a/different/directory` where it wasn't clear from a glance where one path ended and the other started.